lib/dialects/sqlite3/schema/ddl.js

```diff
@@ -43,365 +46,351 @@ class SQLite3_DDL {
   getTableSql() {
     const tableName = this.tableName();
 
-    this.trx.disableProcessing();
-    return this.trx.raw(getTableSql(tableName)).then((result) => {
-      this.trx.enableProcessing();
-      return {
-        createTable: result.filter((create) => create.type === 'table')[0].sql,
-        createIndices: result
-          .filter((create) => create.type === 'index')
-          .map((create) => create.sql),
-      };
-    });
+    return this.client.transaction(
+      async (trx) => {
+        trx.disableProcessing();
+        const result = await trx.raw(getTableSql(tableName));
+        trx.enableProcessing();
+
+        return {
+          createTable: result.filter((create) => create.type === 'table')[0]
+            .sql,
+          createIndices: result
+            .filter((create) => create.type === 'index')
+            .map((create) => create.sql),
+        };
+      },
+      { connection: this.connection }
+    );
   }
 
-  renameTable() {
-    return this.trx.raw(renameTable(this.alteredName, this.tableName()));
+  async isForeignCheckEnabled() {
+    const result = await this.client
+      .raw(isForeignCheckEnabled())
```

Comment:
can we rename one of the methods to make this code less confusing?

Reply:
Sure! I kept the names the same because the functions `createNewTable`, `copyData`, `dropOriginal`, `renameTable` and `getTableSql`, which existed before this PR, are also called exactly the same as their corresponding methods in the `SQLite3_DDL` class. Should I rename all of them?

Reply:
all good then, out-of-scope for this pr